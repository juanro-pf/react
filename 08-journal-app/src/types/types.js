
export const types= {
  login: '[Auth] Login',
  logout: '[Auth] Logout',

  uiSetError: '[UI] Set Error',
  uiUnsetError: '[UI] Unset Error',

  uiStartLoading: '[UI] Start Loading',
  uiFinishLoading: '[UI] Finish Loading',

  notesAddEntry: '[Notes] New Note',
  notesActive: '[Notes] Set Active Note',
  notesLoad: '[Notes] Load Notes',
  notesUpdated: '[Notes] Updated Note',
  notesFileUrl: '[Notes] Updated Image Url',
  notesDelete: '[Notes] Delete Note',
  notesLogoutCleaning: '[Notes] Logout Cleaning'
};