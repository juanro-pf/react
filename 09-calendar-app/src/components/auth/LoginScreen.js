import React from 'react';
import './login.css';
import { useForm } from '../../hooks/useForm';
import { useDispatch } from 'react-redux';
import { startLogin, startRegister } from '../../actions/auth';
import Swal from 'sweetalert2';

export const LoginScreen = () => {

  const dispatch = useDispatch();

  const [ formLoginValues, handleLoginInputChange ]= useForm({
    lEmail: 'jro@gmail.com',
    lPassword: '123456'
  });

  const [ formRegisterValues, handleRegisterInputChange ]= useForm({
    rName: 'Ropo',
    rEmail: 'jropo@gmail.com',
    rPassword: '123456',
    rPassword2: '123456'
  });

  const { lEmail, lPassword }= formLoginValues;
  const { rEmail, rName , rPassword , rPassword2 }= formRegisterValues;

  const handleLogin= (e) => {
    e.preventDefault();
    dispatch( startLogin( lEmail, lPassword ) );
  };

  const handleRegister= (e) => {
    e.preventDefault();

    if( rPassword !== rPassword2 ){
      return Swal.fire('Error', 'Passwords must match', 'error');
    }
    
    dispatch( startRegister( rEmail, rName, rPassword ) );
  };

  return (
    <div className="container login-container">
      <div className="row">
        <div className="col-md-6 login-form-1">
          <h3>Log in</h3>
          <form onSubmit= { handleLogin }>
            <div className="form-group">
              <input 
                type="text"
                className="form-control"
                placeholder="Email"
                name= 'lEmail'
                value= { lEmail }
                onChange= { handleLoginInputChange }
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Password"
                name= 'lPassword'
                value= { lPassword }
                onChange= { handleLoginInputChange }
              />
            </div>
            <div className="form-group">
              <input 
                type="submit"
                className="btnSubmit"
                value="Login" 
              />
            </div>
          </form>
        </div>

        <div className="col-md-6 login-form-2">
          <h3>Register</h3>
          <form onSubmit= { handleRegister }>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Name"
                name= 'rName'
                value= { rName }
                onChange= { handleRegisterInputChange }
              />
            </div>
            <div className="form-group">
              <input
                type="email"
                className="form-control"
                placeholder="Email"
                name= 'rEmail'
                value= { rEmail }
                onChange= { handleRegisterInputChange }
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Password" 
                name= 'rPassword'
                value= { rPassword }
                onChange= { handleRegisterInputChange }
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Repeat password"
                name= 'rPassword2'
                value= { rPassword2 }
                onChange= { handleRegisterInputChange }
              />
            </div>

            <div className="form-group">
              <input 
                type="submit" 
                className="btnSubmit" 
                value="Create account"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  )
}