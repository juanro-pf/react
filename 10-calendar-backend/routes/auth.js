/*
  Rutas de Auth
  host + /api/auth
*/

const { Router }= require('express');
const { check }= require('express-validator');
const { validateFields } = require('../middlewares/fieldValidator');
const { createUser, loginUser, revalidateToken } = require('../controllers/auth');
const { validateJWT }= require('../middlewares/validateJWT');

const router= Router();

router.post(
  '/new',
  [ //middlewares
    check('name', 'Name is required').not().isEmpty(),
    check('email', 'Email is required').isEmail(),
    check('password', 'Password must contain 6 characters').isLength({ min: 6 }),
    validateFields
  ],
  createUser
);

router.post(
  '/',
  [
    check('email', 'Email is required').isEmail(),
    check('password', 'Password must contain 6 characters').isLength({ min: 6 }),
    validateFields
  ],
  loginUser
);

router.get(
  '/renew',
  validateJWT,
  revalidateToken
);

module.exports= router;