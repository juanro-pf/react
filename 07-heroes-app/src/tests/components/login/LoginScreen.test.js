import React from 'react';

import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Testing LoginScreen', () => {

  const historyMock= {
    replace: jest.fn()
  };

  const contextValue= {
    dispatch: jest.fn()
  };

  const wrapper= mount(
    <AuthContext.Provider value={ contextValue }>
      <LoginScreen history={ historyMock }/>
    </AuthContext.Provider>
  );

  test('Snapshot', () => {
    
    expect(wrapper).toMatchSnapshot();

  });

  test('dispatch and navigation', () => {
    
    // Storage.prototype.getItem= '/';

    wrapper.find('button').simulate('click');

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: 'Juan Ro'
      }
    });

    expect(historyMock.replace).toHaveBeenCalledWith('/');

    localStorage.setItem('lastPath', '/dc');

    wrapper.find('button').simulate('click');

    expect(historyMock.replace).toHaveBeenCalledWith('/dc');

  });  

});
