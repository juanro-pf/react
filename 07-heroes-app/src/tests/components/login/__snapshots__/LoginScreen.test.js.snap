// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginScreen Snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
